src/products/utils/validation.ts

```diff
@@ -34,3 +36,14 @@ export const validateProductCreateData = (data: ProductCreateData) => {
 
   return errors;
 };
+
+export const validateVariantData = (
+  data: ProductVariantCreateData | ProductVariantUpdateSubmitData,
+) => {
+  let errors: ProductErrorWithAttributesFragment[] = [];
+
+  if (!data.name) {
+    errors = [...errors, createEmptyRequiredError("name")];
+  }
+  return errors;
+};
```

Comment:
I don't get it, why are you spreading an empty array?
```suggestion
export const validateVariantData = (
  data: ProductVariantCreateData | ProductVariantUpdateSubmitData,
): ProductErrorWithAttributesFragment[] => (
  !data.name ? [createEmptyRequiredError("name")] : [];
);
```